ddl/ddl_worker.go

```diff
@@ -621,12 +621,56 @@ func chooseLeaseTime(t, max time.Duration) time.Duration {
 	return t
 }
 
+// countForErrorOrPanic records the error count for ddl job.
+// 1: err not nil comes from normal error case.
+// 2: err nil comes from abnormal panic case.
+func (w *worker) countForErrorOrPanic(err error, job *model.Job) error {
+	if err == nil {
+		// If run DDL job panic, just cancel the DDL jobs.
+		job.State = model.JobStateCancelling
+		job.ErrorCount++
+
+		// Load global ddl variables.
+		if err1 := loadDDLVars(w); err1 != nil {
+			logutil.Logger(w.logCtx).Error("[ddl] load DDL global variable failed", zap.Error(err1))
+		}
+		errorCount := variable.GetDDLErrorCountLimit()
```

Comment:
line631-637 is duplicated in `if` and `else`, so I think we can put it after line627.